fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -999,29 +999,18 @@ ACTOR Future<Void> getResolution(CommitBatchContext* self) {
 	// Fetch cipher keys if needed.
 	state Future<std::unordered_map<EncryptCipherDomainId, Reference<BlobCipherKey>>> getCipherKeys;
 	if (pProxyCommitData->encryptMode.isEncryptionEnabled()) {
-		static const std::unordered_set<EncryptCipherDomainId> defaultDomainIds = { SYSTEM_KEYSPACE_ENCRYPT_DOMAIN_ID,
-			                                                                        ENCRYPT_HEADER_DOMAIN_ID,
-			                                                                        FDB_DEFAULT_ENCRYPT_DOMAIN_ID };
-		std::unordered_set<EncryptCipherDomainId> encryptDomainIds = defaultDomainIds;
+		std::unordered_set<EncryptCipherDomainId> encryptDomainIds = { SYSTEM_KEYSPACE_ENCRYPT_DOMAIN_ID,
+			                                                           FDB_DEFAULT_ENCRYPT_DOMAIN_ID };
+		if (FLOW_KNOBS->ENCRYPT_HEADER_AUTH_TOKEN_ENABLED) {
+			encryptDomainIds.insert(ENCRYPT_HEADER_DOMAIN_ID);
+		}
 		// For cluster aware encryption only the default domain id is needed
 		if (pProxyCommitData->encryptMode.mode == EncryptionAtRestMode::DOMAIN_AWARE) {
 			for (int t = 0; t < trs.size(); t++) {
 				TenantInfo const& tenantInfo = trs[t].tenantInfo;
 				int64_t tenantId = tenantInfo.tenantId;
 				if (tenantId != TenantInfo::INVALID_TENANT) {
```

Comment:
At this point `mode == cluster_aware`, we should assert `tenantId != invalid_tenant`?

Reply:
Raw access requests will come with invalid tenant id here.

Reply:
My understanding is for `TenantMode::required`, the raw_access is allowed/expected only for `ClearRange` mutations, @sfc-gh-xwang PR if merged should have split the mutations across tenant-boundaries and/or failed the commit request if invalid?

Reply:
Yeah once @sfc-gh-xwang pr is merged we can assert that in `domain_aware` encryption the tenant id is always valid

Reply:
https://github.com/apple/foundationdb/pull/9298#issuecomment-1426898087

Reply:
Btw, the PR linked is `merged`. @sfc-gh-yiwu are you still seeing failure if this code is removed? We may have some more issues to fix in that case..

Reply:
@sfc-gh-yiwu any updates on above discussion?

Reply:
@sfc-gh-yiwu any update on if we can assert here?

Reply:
Actually i just looked at the code the raw access clear mutations are only being split in `postResolution` so it's possible to still have invalid tenant ids here

Reply:
This PR had been rebased onto #9298. As what @sfc-gh-nwijetunga said, clear range split is done post resolution, and so pre-resolution we can still have raw access requests without tenant ids assigned.